contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -105,6 +105,18 @@ While some exceptions may happen, approvers should use the following guidance:
 
 [API changes]: https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md#what-parts-of-a-pr-are-api-changes
 
+Feature lifecycle will remain through the code until post GA or deprecation. For
```

Comment:
This looks great.

Could I make one more small ask?

Would we find a place in this doc to include a table that shows what the steps for introducing a feature look like across the full range of release where the feature must be gated?  I'm imagining a table with columns something like `Kubernetes version`, `Default`, `PreRelease`, `LockedToDefault`, and maybe an additional column that explains what each row means?

Could we also do the same for deprecation?

Reply:
```
RetryGenerateName: {  
  {Version: version.MustParse("1.30"), Default: false, PreRelease: featuregate.Alpha},  
  {Version: version.MustParse("1.31"), Default: true, PreRelease: featuregate.Beta},  
  {Version: version.MustParse("1.32"), Default: true, LockToDefault: true, PreRelease: featuregate.GA},  
},  
```

Isn't this quite self explanatory in terms of feature transitions? Curious on what benefit we'll get from having an additional example in table format. Added an example for deprecation.

Reply:
Good point. I'm happy with this. Thanks for adding the deprecation example.